src/app/_hooks/useToggle.tsx

```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+
+const useToggle = () => {
+  const [isShow, setIsShow] = useState(false);
+
+  const handleClickOpen = () => {
+    setIsShow(true);
+  };
+
+  const handleClickClose = () => {
+    setIsShow(false);
+  };
+
+  const handleClickToggle = () => {
+    setIsShow(!isShow);
+  };
```

Comment:
[ P2 ]

현재 작성해주신 방법같이 isShow라는 값을 직접 사용하여 동작하면 useState의 Setter가 비동기적으로 동작하여 원하는 방향대로 값이 들어가지않을 수 있습니다!

( 어마어마한 광클? )

따라서 해당 부분을
```suggestion
  const handleClickToggle = () => {
    setIsShow(( prevState) => !prevState);
  };
```
와 같이 변경해주시면 무수한 광클이 발생해도 문제없이 동작할 수 있습니다!

Reply:
아하 광클 시 문제가 발생하는군요 ?!
좋은 제안 감사드립니다 !! ㅎㅎ 수정해 놓겠습니다 😊